package com.address.controller;

import java.util.List;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.address.model.dto.AddressDto;
import com.address.model.dto.AddressRequest;
import com.address.service.Impl.AddressService;

@RestController
@RequestMapping("/addresses")
public class AddressController {

	private final AddressService addressService;
	
	
	public AddressController(AddressService addressService) {
		this.addressService = addressService;
	
	}
	
	 // save address for employee
	@PostMapping("/save")
	public ResponseEntity<List<AddressDto>> saveAddress(@RequestBody AddressRequest addressDto){
		List<AddressDto> response = addressService.saveAddress(addressDto);
		return new ResponseEntity<>(response, HttpStatus.CREATED);
	}
	
	// update employee address
	@PutMapping("/update")
	public ResponseEntity<List<AddressDto>> updateAddress(@RequestBody AddressRequest addressDto){
		List<AddressDto> response = addressService.updateAddress(addressDto);
		return new ResponseEntity<>(response, HttpStatus.OK);
	}
	
	//show all address
	@GetMapping("/allAddress")
	public ResponseEntity<List<AddressDto>> getAllAddress(){
		List<AddressDto> response = addressService.getAllAddress();
		return new ResponseEntity<>(response, HttpStatus.OK);
	}
	
	// find address id
	@GetMapping("/{addressId}")
	public ResponseEntity<AddressDto> getAddressById(@PathVariable Long addressId	){
	AddressDto response = addressService.getSingleAddress(addressId);
		return new ResponseEntity<>(response, HttpStatus.OK);
	}
	
	//delete address by emp id or address id
	@DeleteMapping("/delete/{addressId}")
	public ResponseEntity<String> deleteAddress(@PathVariable Long addressId){
		addressService.deleteAddress(addressId);
			return new ResponseEntity<>("Address deleted successfully", HttpStatus.OK);
		}
}









